src/search.js

```diff
@@ -65,10 +72,17 @@ angular
 				 * Updates the model that the user passed in (attributes#param)
 				 */
 				update: function() {
-					scope.search({
-						$param: scope.param
-					});
+					// Do not allow search untill search character meet the minimum length requirement
+					if(scope.minSearchLength && scope.param !== '' && scope.param.length < scope.minSearchLength){
+						scope.searchValidation = true;
+					} else {
+						scope.searchValidation = false;
+						scope.search({
+							$param: scope.param
+						});
+					}
 					scope.dirty = false;
+					scope.$apply();
```

Comment:
If it's just for unit tests, then only add it to your unit tests - not to the source here